### PR TITLE
docs: add commands

### PR DESCRIPTION
**Pull request by mr-cal**
- [X] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

First draft of documenting rockcraft commands.

image

Resolves https://github.com/canonical/rockcraft/issues/101

### CODE REVIEW

File: docs/reference.rst [hunk 1]
```diff
@@ -146,3 +146,51 @@ Example
 
 
 NOTE: at the moment, it is not possible to mix packages and slices in the same stage-packages field.
+
+
+Rockcraft commands
+------------------
+Lifecycle commands
```
Comment (tigarmo) [inline docs/reference.rst:153]: can you screenshot what this new section looks like, rendered? (You might need to update your doc env, there's a new theme)

Comment (mr-cal) [reply docs/reference.rst:153]: Sure thing!

Comment (tigarmo) [reply docs/reference.rst:153]: Thanks! Is there a particular reason why you chose bulleted lists? I think the docs might flow better if we use regular paragraphs for each section, even if they're just a single sentence.
I understand that this is an initial pass to have "something" there, so thanks for doing this regardless ;)

Comment (mr-cal) [reply docs/reference.rst:153]: No particular reason.  I agree that paragraphs would be better.  I updated the docs and the rendered screenshot.

**Review by tigarmo — APPROVED** [review]
thankss

**Comment by mr-cal** [comment]
I see @cjdcordeiro requested a review, so I'm disabling auto-merge to give him a chance to take look when he has time.

**Comment by tigarmo** [comment]
Oh of course, my bad

File: docs/reference.rst [hunk 2]
```diff
@@ -146,3 +146,48 @@ Example
...
+..................
+Lifecycle commands can take an optional parameter ``<part-name>``. When a part name is provided, the command applies to the specific part. When no part name is provided, the command applies to all parts.
+
+clean
+^^^^^
+Removes a part's assets. When no part is provided, the entire build environment (e.g. the LXD instance) is removed
+
+pull
+^^^^
+Downloads or retrieves artefacts defined for each part.
```
Comment (cjdcordeiro) [inline docs/reference.rst:163]:
```suggestion
Downloads or retrieves artifacts defined for each part.
```

File: docs/reference.rst [hunk 3]
```diff
@@ -146,3 +146,48 @@ Example
...
+
+overlay
+^^^^^^^
+Creates part layers over the base filesystem.
```
Comment (cjdcordeiro) [inline docs/reference.rst:167]:
```suggestion
Execute operations defined for each part on a layer over the base filesystem, potentially modifying its contents.
```

File: docs/reference.rst [hunk 4]
```diff
@@ -146,3 +146,48 @@ Example
...
+
+build
+^^^^^
+Builds artefacts defined for each part.
```
Comment (cjdcordeiro) [inline docs/reference.rst:171]:
```suggestion
Builds artifacts defined for each part.
```

File: docs/reference.rst [hunk 5]
```diff
@@ -146,3 +146,48 @@ Example
...
+
+stage
+^^^^^
+Stages built artefacts into a common staging area.
```
Comment (cjdcordeiro) [inline docs/reference.rst:175]:
```suggestion
Stages built artifacts into a common staging area, for each part.
```

File: docs/reference.rst [hunk 6]
```diff
@@ -146,3 +146,48 @@ Example
...
+
+prime
+^^^^^
+Primes artefacts defined for each part.
```
Comment (cjdcordeiro) [inline docs/reference.rst:179]:
```suggestion
Prepare, for each part, the final payload to be packed as a ROCK, performing additional processing and adding metadata files.
```

File: docs/reference.rst [hunk 7]
```diff
@@ -146,3 +146,48 @@ Example
...
+
+pack
+^^^^
+Creates a ROCK from the primed artefacts.
```
Comment (cjdcordeiro) [inline docs/reference.rst:183]:
```suggestion
*This is the default lifecycle command for ``rockcraft``.*

Process parts and create the ROCK as an OCI archive file containing the project payload with the provided metadata.
```

File: docs/reference.rst [hunk 8]
```diff
@@ -146,3 +146,48 @@ Example
...
+
+Other commands
+..............
+init
+^^^^
+Initializes a rockcraft project with a boilerplate ``rockcraft.yaml`` configuration file.
```
Comment (cjdcordeiro) [inline docs/reference.rst:189]:
```suggestion
Initializes a rockcraft project with a boilerplate ``rockcraft.yaml`` file.
```

**Review by cjdcordeiro — CHANGES_REQUESTED** [review]
looking good. thanks

**Pull request merged**